libs/template/materialize.go

```diff
@@ -8,6 +8,7 @@ import (
 const libraryDirName = "library"
 const templateDirName = "template"
 const schemaFileName = "databricks_template_schema.json"
+const metadataFileName = "databricks_template_metadata.json"
```

Comment:
Instead of a separte file, could we just define the order inline in the template schema?

```
{
    "properties": {
        "project_name": {
            "order": 1,
            "type": "string",
            "default": "my_project",
            "description": "Unique name for this project"
        },
        "include_notebook": {
            "order": 2,
            "type": "string",
            "default": "yes",
            "description": "Include a stub (sample) notebook in 'my_project/src'"
        },
}
```

That seems cleaner than a separate file.

And it would have been ideal if the prompts had appeared in an array instead of a map... But ideally. we don't do a breaking change anymore in this stage.

Reply:
The schema file conforms to the JSON schema spec, so only possible if there is some kind of facility in the spec.

Reply:
I think we'd need an extension like the above. Or we should move to an array and accept the breaking change. don't like the idea that we have the complexity of a separate ordering file in perpetuiry, just because we didn't initially think of ordering when we were still in preview.

Reply:
So to recap it seems we have three options:
1. A separate `databricks_template_metadata.json` file. Suffers from discoverability/consistency/redundancy issues, which can only be partially mitigated.
2. An extension of the schema with `order`; we'd need to check if we can do this in a JSON schema compliant way
3. A breaking change, moving from a map to an array. Short-term pain but addresses the long-term issues of #1.

Reply:
Looking at https://json-schema.org/understanding-json-schema/reference/schema.html#vocabularies, we can add extra properties to the schema. To do this properly we'd need to make a copy of the meta schema and include the definition of an `order` field. To do this the quick and dirty way, we can simply add the `order` field to the schema, and let validators ignore it when looking at the input configuration.

Reply:
I implemented this with an `order` field. All logic for property ordering now resides in the `jsonschema` package,.